File: benchmark/cases/sieve/sieve.js
const n = 1000000;
const sieve = new Array(n + 1).fill(true);
sieve[0] = sieve[1] = false;
for (let i = 2; i * i <= n; i++) {
    if (sieve[i]) {
        for (let j = i * i; j <= n; j += i) {
            sieve[j] = false;
        }
    }
}
let count = 0;
for (let k = 2; k <= n; k++) {
    if (sieve[k]) count++;
}
console.log(count);

File: benchmark/cases/sieve/sieve.rs
fn main() {
    let n: usize = 1_000_000;
    let mut sieve = vec![true; n + 1];
    sieve[0] = false;
    sieve[1] = false;
    let mut i = 2;
    while i * i <= n {
        if sieve[i] {
            let mut j = i * i;
            while j <= n {
                sieve[j] = false;
                j += i;
            }
        }
        i += 1;
    }
    let count = sieve.iter().filter(|&&x| x).count();
    println!("{}", count);
}
